sql/catalyst/src/main/java/org/apache/spark/sql/connector/read/SupportsPushDownTopN.java

```diff
@@ -31,8 +31,25 @@
 @Evolving
 public interface SupportsPushDownTopN extends ScanBuilder {
 
+    enum TopNPushedStatus {
```

Comment:
My opinion is that 2 boolean is better than enum in the public API. If we want to add the same feature to limit pushdown, we won't be able to add enum for backward compatibility.